reference/features/security.md

```diff
@@ -0,0 +1,243 @@
+# Security
+
+This guide will teach you how to protect your MeiliSearch instance by setting a master key and how to authorize requests using API keys. You will also learn how to use your master key to create, list, update, and delete API keys with granular permissions.
+
+## Protecting a MeiliSearch instance
+
+**By default, MeiliSearch's API is unprotected**. This means all routes are publicly accessible, and require no authorization to access.
+
+To protect a MeiliSearch instance from unauthorized use, you must supply a master key at launch. This master key can be an alphanumeric string of any length.
+
+Setting up a master key can be done with either command-line options or environment variables. You can read more about [master key configuration in our instance configuration guide](/reference/features/configuration.md#master-key).
+
+:::: tabs
+::: tab CLI
+
+```sh
+./meilisearch --master-key="your_master_key"
+```
+
+:::
+
+::: tab Environment variable
+
+```sh
+export MEILI_MASTER_KEY="your_master_key"
+./meilisearch
+```
+
+:::
+::::
+
+Once you launch MeiliSearch with a master key, all API endpoints except [the health endpoint](/reference/api/health.md#get-health) are automatically protected from unauthorized requests.
+
+From that point on, API requests must include the `Authorization` header to be successful. Read on to learn more.
+
+## Communicating with a protected instance
+
+After an instance is secured, only the [`GET /health` endpoint](/reference/api/health.md) will be publicly available. To access any other API endpoint, you must add a security key with suitable permissions to your request.
+
+It is particularly important to protect an instance when dealing with sensitive data. You can use API keys to ensure only authorized people can search through an index containing medical records:
+
+<CodeSamples id="security_guide_search_key_1" />
+
+### Using default API keys for authorization
+
+When you launch an instance for the first time, MeiliSearch will automatically generate two API keys: `Default Search API Key` and `Default Admin API Key`.
+
+As its name indicates, the `Default Search API Key` can only be used to access [the search route](/reference/api/search.md).
```

Comment:
I was indeed referring to how I called them in the preceding paragraph.

I get your point and `description` would perhaps be more technically correct, but by this point in the text we haven't mentioned API keys have fields. I think it's best to use a generic term like name here to avoid introducing too much stuff at the same time for new users (and experienced programmers will probably not even notice).